sklearn/tree/_criterion.pxd

```diff
@@ -71,6 +57,27 @@ cdef class Criterion:
     ) nogil
     cdef double proxy_impurity_improvement(self) nogil
 
+    cdef void pointer_reset(
```

Comment:
I would actually also rename this to `set_sample_pointers` or something like that.

Reply:
Thank you! Just pushed new commit for this